eZ/Publish/Core/REST/Server/Input/Parser/Criterion/LogicalAnd.php

```diff
@@ -30,15 +35,65 @@ class LogicalAnd extends CriterionParser
      */
     public function parse(array $data, ParsingDispatcher $parsingDispatcher)
     {
-        if (!array_key_exists('AND', $data) && !is_array($data['AND'])) {
-            throw new Exceptions\Parser('Invalid <AND> format');
+        if (!array_key_exists(static::TAG_NAME, $data) || !is_array($data[static::TAG_NAME])) {
+            throw new Exceptions\Parser('Invalid <' . static::TAG_NAME . '> format');
         }
 
         $criteria = array();
-        foreach ($data['AND'] as $criterionName => $criterionData) {
-            $criteria[] = $this->dispatchCriterion($criterionName, $criterionData, $parsingDispatcher);
+
+        $flattenedCriteriaElements = $this->getFlattenedCriteriaData($data[static::TAG_NAME]);
+        foreach ($flattenedCriteriaElements as $criterionElement) {
+            $criteria[] = $this->dispatchCriterion(
+                $criterionElement['type'],
+                $criterionElement['data'],
+                $parsingDispatcher
+            );
+        }
+
+        return new Values\Content\Query\Criterion\LogicalAnd($criteria);
+    }
+
+    /**
+     * @param array $criteriaByType
+     * @return array
+     */
+    protected function getFlattenedCriteriaData(array $criteriaByType)
+    {
+        $criteria = [];
+        foreach ($criteriaByType as $type => $criterion) {
+            if (is_array($criterion) && $this->isNumericArray($criterion)) {
+                foreach ($criterion as $criterionElement) {
+                    $criteria[] = [
+                        'type' => $type,
+                        'data' => $criterionElement,
+                    ];
+                }
+            } else {
+                $criteria[] = [
+                    'type' => $type,
+                    'data' => $criterion,
+                ];
+            }
+        }
+
+        return $criteria;
+    }
+
+    /**
+     * Checks if the given $value is a purely numeric array.
+     *
+     * @param array $value
+     *
+     * @return bool
+     */
+    protected function isNumericArray(array $value)
+    {
+        foreach (array_keys($value) as $key) {
+            if (is_string($key)) {
+                return false;
+            }
         }
 
-        return new LogicalAndCriterion($criteria);
+        return true;
```

Comment:
Following previous comment, body of this method might be:
```php
        reset($value);

        return empty($value) || key($value) === 0;
```

Reply:
This would only tell me if the first element of the array has the index 0 but would also return true with
```
[
  0 => '1234',
  'some_string' => '1234'
]
```

Reply:
But i guess I know what you mean. You mean we don't need to check if the whole array is sequential only having numeric indices, instead just check the first element. Am I right on that?

Reply:
> But i guess I know what you mean. You mean we don't need to check if the whole array is sequential only having numeric indices, instead just check the first element. Am I right on that?

Yes :)